src/core/filesystem.h

```diff
@@ -25,12 +25,65 @@
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #pragma once
 
+#include <dirent.h>
+#include <sys/stat.h>
+#include <unistd.h>
 #include <string>
 #include "google/protobuf/message.h"
 #include "src/core/status.h"
 
 namespace nvidia { namespace inferenceserver {
 
+struct TemporaryDirectory {
+  std::string model_path;
+
+  TemporaryDirectory(std::string path) : model_path(path) {}
```

Comment:
I don't understand this class. It should represent a directory... despite what I said earlier I would call it LocalizedDirectory. It should have a method where you can get the path to the directory (not a model_path... just a path). It should have a way to be constructed "in place" by giving it the path... LocalizedDirectory constructed this way doesn't need to do any cleanup on destruction. Another constructor takes no argument and so it creates a temporary directory and on destruction deletes that directory.

Reply:
The subdirectories are created and managed in the LocalizeFileFolder function. It may not be just a directory but a set of subdirectories as well.

Reply:
That's fine.. LocalizedDirectory just represents the top-level directory

Reply:
The logic for the same is already there in  `S3FileSystem::LocalizeFileFolder()`

Reply:
The LocalizedDirectory object only represents the top-level directory and is used solely to provide that top-level directory path and to know/control when it needs to be deleted. Filling the contents if the directory should stay with the different FileSystem classes.